pkg/handlers/system_intake.go

```diff
@@ -336,3 +336,115 @@ func (h SystemIntakeLifecycleIDHandler) Handle() http.HandlerFunc {
 		}
 	}
 }
+
+// NewSystemIntakeRejectionHandler is a constructor for how we handle
+// rejecting a request
+func NewSystemIntakeRejectionHandler(
+	base HandlerBase,
+	reject func(context.Context, *models.SystemIntake) (*models.SystemIntake, error),
+) SystemIntakeRejectionHandler {
+	return SystemIntakeRejectionHandler{
+		HandlerBase:  base,
+		RejectIntake: reject,
+	}
+}
+
+// SystemIntakeRejectionHandler is the handler for rejecting a SystemIntake
+type SystemIntakeRejectionHandler struct {
+	HandlerBase
+	RejectIntake func(context.Context, *models.SystemIntake) (*models.SystemIntake, error)
+}
+
+type rejectionFields struct {
+	Reason    string `json:"rejectionReason"`
+	NextSteps string `json:"rejectionNextSteps"`
+}
+
+// Handle handles a request for the system intake form
+func (h SystemIntakeRejectionHandler) Handle() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		switch r.Method {
+		case "POST":
+			if r.Body == nil {
+				h.WriteErrorResponse(
+					r.Context(),
+					w,
+					&apperrors.BadRequestError{Err: errors.New("empty request not allowed")},
+				)
+				return
+			}
+			defer r.Body.Close()
+
+			fields := rejectionFields{}
+			if err := json.NewDecoder(r.Body).Decode(&fields); err != nil {
+				h.WriteErrorResponse(r.Context(), w, &apperrors.BadRequestError{Err: err})
+				return
+			}
+
+			// formatting and validation of required inputs
+			intake := &models.SystemIntake{}
+			valFail := false
+			valErr := apperrors.NewValidationError(
```

Comment:
I know we have, in the past, put validation like this in pkg/appvalidations, though that's sort of fallen out of use. Would it be worth trying to move this to there? Also, should validation happen at the handler level? or at the service level?

Reply:
@Igarfinkle  I think you're right that the service should be checking the data points we expect to be there. I will take a stab at moving this code there.

Reply:
My take on this is that the transport layer (aka `handlers`) is the appropriate place to check _structural_ validations (e.g. "this value needs to be non-empty"), whereas the `services` layer is where you would do _logical_ validations (e.g. field X must be non-empty when field Y is set).